typescript/phoenix-sdk/src/market.ts

```diff
@@ -223,36 +294,75 @@ export class Market {
   }
 
   /**
-   * Returns a Phoenix swap transaction
+   * Reloads market data from buffer
    *
-   * @param trader The `PublicKey` of the trader
-   * @param side The side of the order to place (Bid, Ask)
-   * @param inAmount The amount (in whole tokens) of the input token to swap
-   * @param slippage The slippage tolerance (optional, default 0.5%)
-   * @param clientOrderId The client order ID (optional)
+   * @param connection The Solana `Connection` object
+   *
+   * @returns The reloaded Market
    */
-  getSwapTransaction({
-    trader,
-    side,
-    inAmount,
-    slippage = DEFAULT_SLIPPAGE_PERCENT,
-    clientOrderId = 0,
-  }: {
-    trader: PublicKey;
-    side: Side;
-    inAmount: number;
-    slippage?: number;
-    clientOrderId?: number;
-  }) {
-    return getMarketSwapTransaction({
-      marketAddress: this.address,
-      marketData: this.data,
-      trader,
-      side,
-      inAmount,
-      slippage,
-      clientOrderId,
-    });
+  async reloadFromNetwork(connection: Connection): Promise<Market> {
+    const marketData = deserializeMarketData(
+      await connection
+        .getAccountInfo(this.address, "confirmed")
+        .then((accountInfo) => accountInfo?.data)
+    );
+    this.data = marketData;
+    return this;
+  }
+
+  /**
+   * Get a trader's base ATA for a given market
+   *
+   * @param trader The `PublicKey` of the trader account
+   */
+  public getBaseAccountKey(trader: PublicKey): PublicKey {
+    return PublicKey.findProgramAddressSync(
+      [
+        trader.toBuffer(),
+        TOKEN_PROGRAM_ID.toBuffer(),
+        this.data.header.baseParams.mintKey.toBuffer(),
+      ],
+      ASSOCIATED_TOKEN_PROGRAM_ID
+    )[0];
```

Comment:
can we factor this out into a getAta(trader, tokenmint) function?

Reply:
There should actually just be a function for this in spl-token